inventory/environ.py

```diff
@@ -150,13 +150,16 @@ def getSearchHeadClustering(vars_scope):
     shc_vars = vars_scope["splunk"]["shc"]
     shc_vars["label"] = os.environ.get("SPLUNK_SHC_LABEL", shc_vars.get("label"))
     shc_vars["secret"] = os.environ.get("SPLUNK_SHC_SECRET", shc_vars.get("secret"))
+    # Rectify SHC replication factor
+    sh_count = len(inventory.get("splunk_search_head", {}).get("hosts", []))
+    shc_vars["replication_factor"] = min(sh_count, int(shc_vars.get("replication_factor", 0)))
```

Comment:
This may be a good spot to raise an exception to let users know the user-defined replication factor cannot be met instead of just correcting it for them?